### DISCUSSION TITLE
React Wrapper Library for Photo-Sphere-Viewer

### DISCUSSION THREAD

**Discussion by Elius94** [mistic100/Photo-Sphere-Viewer#749] [Show and tell]
## Introducing React.js Wrapper for Photo-Sphere-Viewer

Hi mistic100! Congratulations on your development work for this wonderful library.

I am writing to introduce you to my wrapper-lib for your JS library.
It can handle a good percentage of the features of the original library used as pure Javascript, including all plugins and adapters.
Before I released the new September version with the LittlePlanet Mode, I had created this effect by handling the zoom and wide angle settings in realtime. In addition, by clicking on the globe you can 'drop' into the normal view. Like the spherical photos of DJI drones.

[](https://user-images.githubusercontent.com/14907987/180257570-4070d0b4-b3d8-493d-8f23-efde84074573.mp4)

You can find all directions in the Readme.

If you like the project you might consider mentioning it somewhere for those who are interested in using your lib under the React environment?

Thanks, I hope you like my work.

Here you can find my repo: https://github.com/Elius94/react-photo-sphere-viewer/

And the NPM package: https://www.npmjs.com/package/react-photo-sphere-viewer

**Comment by mistic100** [comment]
@Elius94 yes I will add a section in the doc for people to list their framework integrations

Your little planet is very cool, this was actually my goal to be able to transition to normal view. My first try was not very successfull (https://photo-sphere-viewer.js.org/demos/basic/6-animation.html). Can I borrow your work for this feature ?

**Reply by Elius94** [reply]
Thank you very much! 

> Can I borrow your work for this feature?

Nice! For me it's ok, but can you add me as a contributor? It will be a pleasure 😉

**Reply by mistic100** [reply]
The page is live https://photo-sphere-viewer.js.org/guide/frameworks.html

---

I don't have a specific list of contributors beside the [github page](https://github.com/mistic100/Photo-Sphere-Viewer/graphs/contributors) if you make a PR. Or for specific releases [like 4.7.1](https://github.com/mistic100/Photo-Sphere-Viewer/releases/tag/4.7.1). What did you have in mind ?

**Reply by Elius94** [reply]
Hi! nice page, I will link it in my readme! Thanks.
Oh yes, I think to be on the contributors list I should at least add an issue with the modified code. I'll think about it but if it's a problem go ahead

**Reply by mistic100** [reply]
I looked at your code and you use in fact exactly the same technique as in https://photo-sphere-viewer.js.org/demos/basic/6-animation.html  
Only with a lower fisheye value.

In wide displays you have the same problem as I have : black patches in the corners because the camera is too far from the sphere mesh. 

That's what I was able to fix with a custom shader https://github.com/mistic100/Photo-Sphere-Viewer/blob/dev/src/adapters/little-planet/index.js#L126-L152 (not my code). But I don't know how to seamless transition from this to a standard view.

**Reply by Elius94** [reply]
it's true! I too had the problem on higher resolution screens. In my opinion one thing you could do is to sample the colour of the outer part of the image and make a background with the same colour
